test.js

```diff
@@ -110,3 +153,96 @@ test('custom client', t => {
     })
   })
 })
+
+test('custom client inside a namespace', (t) => {
+  t.plan(7)
+  const fastify = Fastify()
+  const redis = require('redis').createClient({ host: 'localhost', port: 6379 })
+
+  fastify.register(fastifyRedis, {
+    namespace: 'test',
+    client: redis
+  })
+
+  fastify.ready((err) => {
+    t.error(err)
+    t.is(fastify.redis.test, redis)
+
+    fastify.redis.test.set('key', 'value', (err) => {
+      t.error(err)
+      fastify.redis.test.get('key', (err, val) => {
+        t.error(err)
+        t.equal(val, 'value')
+
+        fastify.close(function (err) {
+          t.error(err)
+          fastify.redis.test.quit(function (err) {
+            t.error(err)
+          })
+        })
+      })
+    })
+  })
+})
+
+test('fastify.redis.test should throw with duplicate connection namespaces', (t) => {
+  t.plan(1)
+
+  const namespace = 'test'
+
+  const fastify = Fastify()
+  t.teardown(() => fastify.close())
+
+  fastify
+    .register(fastifyRedis, {
+      host: '127.0.0.1',
+      namespace
+    })
+    .register(fastifyRedis, {
+      host: '127.0.0.1',
+      namespace
+    })
+
+  fastify.ready((err) => {
+    t.is(err.message, `Redis '${namespace}' instance namespace has already been registered`)
+  })
+})
+
+test('Should throw when trying to register multiple instances without giving a namespace', (t) => {
+  t.plan(1)
+
+  const fastify = Fastify()
+  t.teardown(() => fastify.close())
+
+  fastify
+    .register(fastifyRedis, {
+      host: '127.0.0.1'
+    })
+    .register(fastifyRedis, {
+      host: '127.0.0.1'
+    })
+
+  fastify.ready((err) => {
+    t.is(err.message, 'fastify-redis has already been registered')
+  })
+})
+
+test('Should throw when trying to register multiple instances with different scopes', (t) => {
+  t.plan(1)
+
+  const fastify = Fastify()
+  t.teardown(() => fastify.close())
+
+  fastify
+    .register(fastifyRedis, {
+      host: '127.0.0.1'
+    })
+    .register(fastifyRedis, {
+      host: '127.0.0.1',
+      namespace: 'test'
+    })
+
+  fastify.ready((err) => {
+    t.is(err.message, 'A Redis instance has already been registered without a namespace')
+  })
+})
```

Comment:
Sorry, I didn't explain well my point, this is what I intended for 👍
So we can test that a user can start using this new feature without breaking some others plugins

```suggestion
})

test('Should not throw in with different contexts', (t) => {
  t.plan(1)

  const fastify = Fastify()
  t.teardown(() => fastify.close())

  fastify.register(function (instance, options, next) {
    fastify.register(fastifyRedis, {
      host: '127.0.0.1'
    })
    next()
  })

  fastify.register(function (instance, options, next) {
    fastify
      .register(fastifyRedis, {
        host: '127.0.0.1',
        namespace: 'test1'
      })
      .register(fastifyRedis, {
        host: '127.0.0.1',
        namespace: 'test2'
      })
    next()
  })

  fastify.ready(t.error)
})
```

Reply:
Oh ok I see now :rofl: